kernel/packages/gif-processor/worker.ts

```diff
@@ -1,85 +1,125 @@
-    if (hasToBeResized) {
-      let scalingFactor = (gifCanvas.width > gifCanvas.height) ? (gifCanvas.width / maxGIFDimension) : (gifCanvas.height / maxGIFDimension)
-      resizedCanvas.width = gifCanvas.width / scalingFactor
-      finalWidth = resizedCanvas.width
-
-      resizedCanvas.height = gifCanvas.height / scalingFactor
-      finalHeight = resizedCanvas.height
+  async function DownloadAndProcessGIF(e: ProcessorMessage) {
+    abortController = new AbortController()
+    const signal = abortController.signal
+
+    try {
+      const imageFetch = fetch(e.data.url, { signal })
+      const response = await imageFetch
+      abortController = null
+
+      const buffer = await response.arrayBuffer()
+      const parsedGif = await parseGIF(buffer)
+      const decompressedFrames = decompressFrames(parsedGif, true)
+      const frameDelays = new Array()
+      const framesAsArrayBuffer = new Array()
+      let hasToBeResized = false
+
+      frameImageData = undefined
+
+      gifCanvas.width = decompressedFrames[0].dims.width
+      let finalWidth = gifCanvas.width
+
+      gifCanvas.height = decompressedFrames[0].dims.height
+      let finalHeight = gifCanvas.height
+
+      hasToBeResized = gifCanvas.width > maxGIFDimension || gifCanvas.height > maxGIFDimension
+      if (hasToBeResized) {
+        let scalingFactor =
+          gifCanvas.width > gifCanvas.height ? gifCanvas.width / maxGIFDimension : gifCanvas.height / maxGIFDimension
+        resizedCanvas.width = gifCanvas.width / scalingFactor
+        finalWidth = resizedCanvas.width
+
+        resizedCanvas.height = gifCanvas.height / scalingFactor
+        finalHeight = resizedCanvas.height
+      }
+
+      for (const key in decompressedFrames) {
+        frameDelays.push(decompressedFrames[key].delay)
+
+        const processedImageData = GenerateFinalImageData(decompressedFrames[key], hasToBeResized)
+        if (processedImageData) framesAsArrayBuffer.push(processedImageData.data.buffer)
+      }
+
+      self.postMessage({
+        success: true,
+        arrayBufferFrames: framesAsArrayBuffer,
+        width: finalWidth,
+        height: finalHeight,
+        delays: frameDelays,
+        url: e.data.url,
+        id: e.data.id
+      } as WorkerMessageData)
+    } catch (err) {
+      abortController = null
+      self.postMessage({
+        success: false,
+        id: e.data.id
+      } as Partial<WorkerMessageData>)
     }
-
-    for (const key in decompressedFrames) {
-      frameDelays.push(decompressedFrames[key].delay)
-
-      const processedImageData = GenerateFinalImageData(decompressedFrames[key], hasToBeResized)
-      framesAsArrayBuffer.push(processedImageData.data.buffer)
-    }
-
-    self.postMessage({
-      arrayBufferFrames: framesAsArrayBuffer,
-      width: finalWidth,
-      height: finalHeight,
-      delays: frameDelays,
-      id: e.data.id,
-    }, framesAsArrayBuffer)
```

Comment:
We should still pass this second parameter of the `self.postMessage()` (framesAsArrayBuffer) if we want the data to be passed "by reference".

Can you try adding that to your `self.postMessage()` call? if it doesn't break anything then it works, otherwise I guess it's fine